packages/graphql-validate-fixtures/src/index.ts

```diff
@@ -1,68 +1,181 @@
 import {readFile, readJSON} from 'fs-extra';
-import {isAbsolute, resolve} from 'path';
+import {resolve} from 'path';
+import * as glob from 'glob';
+import {Source, parse, concatAST, GraphQLSchema} from 'graphql';
+import {getGraphQLConfig, GraphQLProjectConfig} from 'graphql-config';
+import {AST, compile, Operation} from 'graphql-tool-utilities/ast';
 import {
-  GraphQLSchema,
-  buildClientSchema,
-  Source,
-  parse,
-  concatAST,
-} from 'graphql';
-import {compile} from 'graphql-tool-utilities/ast';
+  getGraphQLFilePath,
+  getGraphQLProjects,
+} from 'graphql-tool-utilities/config';
 
 import {
+  getOperation,
   validateFixtureAgainstAST,
-  validateFixtureAgainstSchema,
   Validation,
   Fixture,
+  MissingOperationError,
 } from './validate';
 
-export interface Paths {
-  fixturePaths: string[];
-  operationPaths?: string[];
-  schemaPath: string;
-}
-
 export interface Options {
-  schemaOnly?: boolean;
+  cwd: string;
 }
 
 export interface Evaluation extends Validation {
-  fixturePath: string;
   scriptError?: Error;
 }
 
 export async function evaluateFixtures(
-  {fixturePaths, operationPaths = [], schemaPath}: Paths,
-  {schemaOnly = false}: Options = {},
+  fixturePaths: string[],
+  {cwd}: Options,
 ): Promise<Evaluation[]> {
-  let schema: GraphQLSchema;
+  const config = getGraphQLConfig(resolve(cwd));
 
-  try {
-    const schemaJSON = await readJSON(schemaPath, {encoding: 'utf8'});
-    schema = buildClientSchema(schemaJSON.data);
-  } catch (error) {
-    throw new Error(
-      `Error parsing '${schemaPath}':\n\n${error.message.replace(
-        /Syntax Error GraphQL \(.*?\) /,
-        '',
-      )}`,
-    );
+  const projectOperations = await Promise.all(
+    getGraphQLProjects(config).map(getOperationsForProject),
+  );
+
+  return runForEachFixture(fixturePaths, (fixture) =>
+    evaluateFixture(fixture, projectOperations),
+  );
+}
+
+interface FixtureOperation {
+  ast: AST;
+  error?: any;
+  operation: Operation;
+  operationName: string;
+}
+
+interface FixtureOperationError {
+  ast: AST;
+  error: any;
+  operation?: Operation;
+  operationName?: string;
+}
+
+function evaluateFixture(
```

Comment:
This flow feels really roundabout for me, or maybe it is just not sufficiently broken down into functions. I would expect it to be more something like this (pseudocode):

```ts
const projectsWithMatchingOperation = findMatchingOperations(fixture, projects);

if (projects.length === 0) {
  return noMatchingOperationsError(fixture); // if we want, we can do a fancy thing here to detect operations that had a close name, but didn't match. Might be overkill, I think for now just say we couldn't find any matching ones is enough
} if (projects.length > 1) {
  return multipleMatchingOperationsError(fixture, projectsWithMatchingOperation); // Tell them how to resolve
}

return validateAgainstProject(fixture, projectsWithMatchingOperation[0]);
```

Reply:
I'll take another pass at it to clear up the flow.

Reply:
I just finished a pretty crazy rewrite to address this. You probably want to have another look again tomorrow. The flow is much more aligned with what you had in mind. There may be some more tweaks to make yet, but i think we're much closer now.